navigation.py

```diff
@@ -0,0 +1,71 @@
+# Navigation module for ThatsJustTheWazeItIs
+from collections import defaultdict
+import json
+import threading
+from map import Graph
+
+class Navigation:
+    """docstring for ."""
+
+    def __init__(self, filename):
+        self.graph = Graph(filename)
+        self.__roomEmpty = threading.Lock()
+        self.__turnstile = threading.Lock()
```

Comment:
It looks like every time the map is accessed for directions a car moves from its current street to the next one in its path (so a write). Are there any reader threads in this solution? If not, we can just protect the map state with a single mutex.